File: java/gol.java

import java.util.concurrent.TimeUnit;

class Gol {
  public static int alive(int[][] matrix, int x, int y) {
    int[][] dirs = {
        {1, 0},
        {1, 1},
        {1, -1},
        {-1, -1},
        {-1, 0},
        {-1, 1},
        {0, 1},
        {0, -1},
    };
    int rows = matrix.length;
    int cols = matrix[0].length;
    int count = 0;
    for (int[] dir : dirs) {
      int newx = (x + dir[0] + cols) % cols;
      int newy = (y + dir[1] + rows) % rows;
      if (matrix[newy][newx] == 1) {
        count++;
      }
    }
    return count;
  }

  public static void display(int[][] matirx) {
    System.out.print("\u001B[1J\u001B[10;10H\n");
    for (int[] row : matirx) {
      for (int col : row) {
        if (col == 1) {
          System.out.print("⬜");
        } else {
          System.out.print("⬛");
        }
      }
      System.out.println("");
    }
  }

  public static int[][] next_step(int[][] matrix) {
    int rows = matrix.length;
    int cols = matrix[0].length;
    int[][] buffer = new int[rows][cols];
    for (int i = 0; i < rows; i++) {
      for (int j = 0; j < cols; j++) {
        int alives = alive(matrix, j, i);
        if (alives == 3) {
          buffer[i][j] = 1;
        } else if (alives == 2) {
          buffer[i][j] = matrix[i][j];
        } else {
          buffer[i][j] = 0;
        }
      }
    }
    return buffer;
  }

  public static void main(String[] args) {
    int[][] matrix = {
        {0, 0, 0, 0, 0},
        {0, 0, 1, 0, 0},
        {0, 0, 0, 1, 0},
        {0, 1, 1, 1, 0},
        {0, 0, 0, 0, 0},
    };

    for (int i = 0; i >-1; i++) {
      Gol.display(matrix);
      matrix = Gol.next_step(matrix);
      try {
        Thread.sleep(200);
      } catch (InterruptedException e) {
      }
    }
  }
}

File: swift/gol.swift
import Foundation
typealias MatLike = Array<Array<Int>>

func matdis(matrix:MatLike) {
    print("\u{001b}[1J\u{001b}[1;1H")
    for row in matrix {
        for col in row {
            if col == 1 {
                print("⬜", terminator:"")
            }else{
                print("⬛", terminator:"")
            }
        }
        print()
    }
}


func alive(matrix:MatLike,x:Int,y:Int) -> Int {
    var count = 0
    for dy in -1...1 {
        for dx in -1...1 {
            if (dx == 0 && dy == 0 ){
                continue
            }
            let newx = (x+matrix[0].count+dx) % matrix[0].count
            let newy = (y+matrix.count+dy) % matrix.count
            if (matrix[newy][newx] == 1){
                count+=1
            }
       }
    }
    return count
}


func next_step(matrix:MatLike) -> MatLike {
    var buffer = Array(repeating: Array(repeating: 0,count: matrix[0].count),count: matrix.count)
    for y in 0..<matrix.count {
        for x in 0..<matrix[0].count{
            let alive_count = alive(matrix:matrix,x:x,y:y)
            if ( alive_count == 3 ){
                buffer[y][x] = 1
            }else if alive_count==2 {
                buffer[y][x] = matrix[y][x]    
            }else {
                buffer[y][x] = 0
                    
            }
        }
    }
    return buffer

}


var matrix = [
        [0,0,0,0,0],
        [0,0,1,0,0],
        [0,0,0,1,0],
        [0,1,1,1,0],
        [0,0,0,0,0],
]


for _ in 0...2000{
    matdis(matrix:matrix)
    matrix = next_step(matrix:matrix)
    Thread.sleep(forTimeInterval: 0.05)
}